clang/include/clang/AST/OpenACCClause.h

```diff
@@ -156,6 +156,64 @@ class OpenACCSelfClause : public OpenACCClauseWithCondition {
                                    Expr *ConditionExpr, SourceLocation EndLoc);
 };
 
+/// Represents one of a handful of classes that have integer expressions.
+/// Semantically, many only permit a single expression, with a few that permit
+/// up to 3.
+class OpenACCClauseWithIntExprs : public OpenACCClauseWithParams {
+  llvm::SmallVector<Expr *> IntExprs;
```

Comment:
Use tail-allocation instead

Reply:
I moved this closer to tail-allocation.  I didn't see value in doing this with tail-allocation in THIS case because it is just a single `Expr`.  However, I changed this class to work correctly when I inherit from it directly with 'trailing' storage.

Reply:
Why do you need a vector then?

Reply:
There are OTHER clauses coming in the future that use multiple `Exprs`, so I'm designing the class hierarchy to handle that as well.  The intent is for THIS class to be the base for all that have `IntExprs`, but the one I am implementing NOW is only a 'single' Expr*.

So my intent is that I WILL use the trailing-storage on the ones that take multiples (and have now done the changes so that will be possible, by making this an arrayref), but as an optimization, just use a DataMember for the 'special cases' where there is only 1.

Reply:
Better to have here just a single expr and later just rework it for trailing storage, I think.

Reply:
Ok, removed the type, so we now just store the info in the 'single expr' base class and reference it from there.